ruff.toml

```diff
@@ -0,0 +1,86 @@
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2025 Felix Fontein <felix@fontein.de>
+
+line-length = 160
+
+[lint]
+# https://docs.astral.sh/ruff/rules/
+
+select = ["A", "B", "E", "F", "FA", "FLY", "UP", "SIM"]
+ignore = [
+    # Better keep ignored (for now)
+    "F811",  # Redefinition of unused `xxx` (happens a lot for fixtures in unit tests)
+    "E402",  # Module level import not at top of file
+    "E741",  # Ambiguous variable name
+    "UP012",  # unnecessary-encode-utf8
+    "UP015",  # Unnecessary mode argument
+    "SIM105",  # suppressible-exception
+    "SIM108",  # if-else-block-instead-of-if-exp
+    # To fix later:
+    "B905",  # zip-without-explicit-strict - needs Python 3.10+
+    # To fix:
+    "F401",  # Unused import
+    "E721",  # Type comparison
+    "E713",  # "not in" test
+    "E714",  # "not is" test
+    "F841",  # Unused variable
+    "UP004",  # Class `xxx` inherits from `object`
+    "UP006",  # Use `dict` instead of `t.Dict` for type annotation
+    "UP007",  # Use `X | Y` for type annotations
+    "UP008",  # Use `super()` instead of `super(__class__, self)`
+    "UP009",  # UTF-8 encoding declaration is unnecessary
+    "UP014",  # Convert `xxx` from `NamedTuple` functional to class syntax
+    "UP018",  # Unnecessary `str` call (rewrite as a literal)
+    "UP020",  # Use builtin `open`
+    "UP024",  # Replace aliased errors with `OSError`
+    "UP025",  # Remove unicode literals from strings
+    "UP028",  # Replace `yield` over `for` loop with `yield from`
+    "UP029",  # Unnecessary builtin import: `open`
+    "UP030",  # Use implicit references for positional format fields
+    "UP031",  # Use format specifiers instead of percent format
+    "UP032",  # Use f-string instead of `format` call
```

Comment:
We should have a couple of those, yes, but non-trivial cases. I believe ruff can fix it, right? I am curious what the fux would look like.

Reply:
Yep, that will be interesting. My plan is that once this is merged, to start a new PR where I will uncomment a few and let ruff fix them, and see what happens...